test/cmdlineargs.jl

```diff
@@ -248,53 +248,66 @@ let exename = `$(Base.julia_cmd()) --precompiled=yes --startup-file=no`
     # tested in test/parallel.jl, test/examples.jl)
     @test !success(`$exename --worker=true`)
 
-    escape(str) = replace(str, "\\", "\\\\")
-
     # test passing arguments
     let testfile = tempname()
         try
             # write a julia source file that just prints ARGS to STDOUT
             write(testfile, """
                 println(ARGS)
-            """)
+                """)
             @test readchomp(`$exename $testfile foo -bar --baz`) ==
                 "String[\"foo\", \"-bar\", \"--baz\"]"
             @test readchomp(`$exename $testfile -- foo -bar --baz`) ==
                 "String[\"foo\", \"-bar\", \"--baz\"]"
             @test readchomp(`$exename -L $testfile -e 'exit(0)' -- foo -bar --baz`) ==
                 "String[\"foo\", \"-bar\", \"--baz\"]"
             @test split(readchomp(`$exename -L $testfile $testfile`), '\n') ==
-                ["String[\"$(escape(testfile))\"]", "String[]"]
+                ["String[]", "String[]"]
             @test !success(`$exename --foo $testfile`)
             @test readchomp(`$exename -L $testfile -e 'exit(0)' -- foo -bar -- baz`) == "String[\"foo\", \"-bar\", \"--\", \"baz\"]"
         finally
             rm(testfile)
         end
     end
 
-    # test the script name
-    let a = tempname(), b = tempname()
-        try
-            write(a, """
-                println(@__FILE__)
-                println(PROGRAM_FILE)
-                println(length(ARGS))
```

Comment:
No. Originally I added this since the number of arguments could vary depending on when we would run the check. I could still include this check but it would only be for verifying that the `ARGS` were no longer includes the `PROGRAM_FILE` as `ARGS[1]`